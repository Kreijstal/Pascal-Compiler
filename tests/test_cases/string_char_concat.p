program string_char_concat;
var
  source: string;
  prefix: string;
  suffix: string;
  idx: integer;
begin
  source := 'ABCDE';
  prefix := '';
  suffix := '';
  idx := 2;

  prefix := prefix + source[idx];
  suffix := source[idx + 1] + suffix;

  WriteLn('prefix=' + prefix);
  WriteLn('suffix=' + suffix);
  WriteLn('pair=' + (source[idx] + source[idx + 1]));
  WriteLn('roundtrip=' + (prefix + suffix));
end.
