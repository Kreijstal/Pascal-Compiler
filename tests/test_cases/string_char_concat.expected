prefix=B
suffix=C
pair=BC
roundtrip=BC
